src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java

```diff
@@ -97,17 +104,20 @@ private boolean checkSharpness(ItemStack stack) {
         }
 
         NBTTagList enchants = (NBTTagList) stack.getTagCompound().getTag("ench");
-		
-		if (enchants == null) {
-			return false;
-		}
-			
+
+        // IntelliJ marks (enchants == null) as always false but this is not correct.
+        // In case of a Hotbar Slot without any Enchantment on the Stack it contains,
+        // this will throw a NullPointerException if not accounted for!
+        //noinspection ConstantConditions
+        if (enchants == null) {
+            return false;
+        }
 
         for (int i = 0; i < enchants.tagCount(); i++) {
-            NBTTagCompound enchant = ((NBTTagList) enchants).getCompoundTagAt(i);
+            NBTTagCompound enchant = enchants.getCompoundTagAt(i);
             if (enchant.getInteger("id") == 16) {
                 int lvl = enchant.getInteger("lvl");
-                if (lvl >= 16) {
+                if (lvl >= 42) { // dia sword against full prot 5 armor is deadly somehere >= 34 sharpness iirc
```

Comment:
Where does `42` come from then?

Reply:
It was a number we decided on. Not everyone will have exactly 32k and some people want to use it on minigame servers  that have stuff like 100 and 1000 sharpness

Reply:
Also 127s and 1337s exist on 2b. Just low enough it doesn't use useless tools

Reply:
![42](https://i.imgur.com/6Ifo6HF.png)